lms/djangoapps/experiments/flags.py

```diff
@@ -128,6 +135,13 @@ def get_bucket(self, course_key=None, track=True):
             # We need username for stable bucketing and id for tracking, so just skip anonymous (not-logged-in) users
             return 0
 
+        if is_masquerading(request.user, course_key) and is_masquerading_as_specific_student(request.user, course_key):
+            user = get_specific_masquerading_user(request.user, course_key)
+            masquerade_active = True
+        else:
+            user = request.user
+            masquerade_active = False
```

Comment:
`masquerade_active` is a bit misleading to me. If I didn't read look at how it was defined, I would assume it would be true for *any* masquerading, not just specific-user masquerading.

Reply:
Good call, will fix.